### ISSUE TITLE
awk syntax error on RHEL7

### ISSUE THREAD

**Issue by rconde01** [coin-or/coinbrew#83]
running:
```
./coinbrew fetch Ipopt:releases/3.14.12 --no-prompt
```
on RHEL7 clones Ipopt successfully but on getting dependencies gives:
```
awk: cmd. line:1: /^stable\\/2.0{print;}
awk: cmd. line:1:               ^ syntax error
```
It works fine on RHEL8.

RHEL7 bash version: 4.2.46
RHEL7 awk version: 4.0.2
RHEL8 bash version: 4.4.20
RHEL8 awk version: 4.2.1

**Comment by rconde01** [comment]
Yes, this is the most recent version of coinbrew. The error comes from https://github.com/coin-or/coinbrew/blob/55cda7f976461c4d1daa3dc2a71dc9783785a901/coinbrew#L1235

Here's the log: [log.txt](https://github.com/coin-or/coinbrew/files/11335170/log.txt)

Another test which might be useful:
```
version_num=stable/2.0
echo "/^${version_num/\//\\\/}/{print;}"
```
RHEL8 `/^stable\/2.0/{print;}`
RHEL7 `/^stable\\/2.0/{print;}`

So I think the issue is differences in the shell.

In discussing with others it has suggested it should be:
```
"/^${version_num/\//\\/}/{print;}"`
```
rather than
```
"/^${version_num/\//\\\/}/{print;}"   
```

For the test above, this gives consistent results with the original RHEL8 result on both RHEL7 and RHEL8. However, I'm not sure it handles the possible values of `version_num`. As an aside, few people also said that is not the "proper" way to use variables with awk, but this is out of my wheelhouse.

**Comment by tkralphs** [comment]
Ah, OK, thanks for running this down. I'm honestly not an expert shell scripter, so I'm not surprised if I did something that is not "proper" :). What that chunk of code is doing is finding the latest release corresponding to a given stable version. So in principle, `version_num` always be of the form `stable/x.y` or just `x.y`. It's getting just the `x.y` part and looking for a release tag `releases/x.y.z` with max value of `z`. So I guess your fix should be fine. 

@svigerske, you are more of an expert than me, what do you think?

**Comment by svigerske** [comment]
I understood `awk "/^${version_num/\//\\\/}/{print;}"` as "print every line starting with `$version_num`". The `/\//\\/` is to escape forward slashes in `$version_num`. I would think you get the same via grep, with the advantage that it doesn't need forward slashes to be escaped.

A `${version_num: -3}` gives the last three characters from `${version_num}`. I'm not sure that he wants the `stable/` to be removed. Maybe, but then there would also be a problem if x > 9 or y > 9.

**Comment by rconde01** [comment]
maybe it's a bug in `version_num`...the name implies it's just the number

**Comment by tkralphs** [comment]
OK, sorry for this. I was a bit too hasty in my earlier answer. Now that I've taken a closer look, yes, it's a bug in `version_num`. Indeed, it should be a number. I was not properly stripping off `stable/` when reading `config.yml`. It's never come up in any of the projects I manage because a while back, I switched the version number string in most of the YAML files to be either `x.y` or `x.y.z` rather than `stable/x.y` or `releases/x.y.z` but at least in this Ipopt release, it is still in the old form.  

Actually, now that I dug in further, it turns out this was already reported previously in #68! But the fix that we cam up with was exactly what caused the syntax error here and didn't get to the root cause. After the previous fix (and on @rconde01's RHEL8 machine apparently), it seems that the stable branches of all dependencies would be checked out instead of getting the latest release, which was the actual intent. So the error was fixed, but the script was still not working an intended. At that time, I think I didn't notice this because it just so happened that ASL 2.0 had no releases and so getting the stable version was proper behavior and there was a fix that actually ensured this would happen (see 112519). So all of this combined to make this bug slip under the radar.

In any case, I was a bit incorrect about what the code snippet that actually errored was doing. Indeed, `version_num` was supposed to be a number, so all it was actually doing was taking the list of tags and filtering it down to ones of the form releases/x.y.*, where `x.y` is the `version_num`, then sorting to find the latest.

So with all that said, I don't actually think we do need this
```
/^${version_num/\//\\\/}/{print;}
```
It seems that
```
/^${version_num}/{print;}
```
should be fine and this what was originally there, actually. @svigerske suggested adding `/\//\\\` as a fix for #68, but I take the full blame for not putting my finger on exactly what was happening at that time.

**Comment by svigerske** [comment]
> OK, sorry for this. I was a bit too hasty in my earlier answer. Now that I've taken a closer look, yes, it was a bug in `version_num`. Indeed, it should be a number. I was not properly stripping off `stable/` when reading `config.yml`. It's never come up in any of the projects I manage because a while back, I switched the version number string in most of the YAML files to be either `x.y` or `x.y.z` rather than `stable/x.y` or `releases/x.y.z` but at least in this Ipopt release, it is still in the old form.

So I should remove the `stable/` in https://github.com/coin-or/Ipopt/blob/stable/3.14/.coin-or/config.yml ?

> @svigerske suggested adding `/\//\\\` as a fix for #68, but I take the full blame for not putting my finger on exactly what was happening at that time.

It seems to be just a year ago, but I have no memory of this... :(

**Comment by tkralphs** [comment]
> So I should remove the stable/ in https://github.com/coin-or/Ipopt/blob/stable/3.14/.coin-or/config.yml ?

Yes, the `stable/` is not needed and I personally think it's better to just use the actual version number there. If it's two digits, it's automatically interpreted as a stable branch. The script can handle either case, though.  

> It seems to be just a year ago, but I have no memory of this... :(

I know how you feel :(. I looked back to see how that regex got added in the first place, since it doesn't look like something I would concoct and was really surprised to see that we had already tried to fix this bug once :).
